bpf/bpf_sock.c

```diff
@@ -347,6 +404,15 @@ static __always_inline int __sock4_xlate_fwd(struct bpf_sock_addr *ctx,
 		return -ENOENT;
 	}
 
+#ifndef ENABLE_IPV6
```

Comment:
Does this mean that if you simply enable IPv6, then it will break the connectivity for IPv4?

Even if the KIAM metadata server may not support IPv6 for some time, I don't think that precludes running IPv6 applications over the network for other purposes.

Wondering if you considered passing down the IPv4 vs. IPv6 context from the caller (as bool for example) so that we just skip this step for the sock6_xlate_v4_in_v6() case?

Reply:
Sure, that sounds like a better solution to me.